### PR TITLE
bump persistenceagent rock to ckf-1.8 (kfp 2.0.3)

### LINKED ISSUE #47: Update rockraft file of persistenceagent for CKF 1.8
### What needs to get done

Update the corresponding `rockcraft.yaml` file to make sure the version of CKF 1.8 is built. 

DoD:
- The `rockcraft.yaml` is updated
- There is automation that builds the image in each PR
- There is automation that pushes the image, once a PR is merged

### Why it needs to get done

To make sure we have less CVEs in 1.8 than upstream, and enable us to be able to potentially create patches for package versions in the future

### PR DESCRIPTION
**Pull request by orfeas-k**
- bump source version from `2.0.0-alpha.7` to `2.0.3`
- update according to canonical/bundle-kubeflow#747
- rename base to ubuntu@20.04 as per new `rockcraft` syntax
- use `1.20/stable` snap for Go
- refactor `tox.ini` according to canonical/oidc-authservice-rock#14 and canonical/bundle-kubeflow#763
- use placeholder instead of integration tests (they fail for the same reasons as https://github.com/canonical/kfp-operators/issues/377)
- add placeholder instead of sanity tests since there are none at the moment

Note CI needs a `sanity` and `integration` environment in order to not fail and be able to publish the produced ROCK after merging this.

### Testing
ROCK can be built and unit goes to active using it. I didn't run any hands-on tests since bundle-integration and manual tests will be run once all kfp ROCKs are updated.

Closes #47

### CODE REVIEW

File: persistenceagent/tox.ini [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [tox]
 skipsdist = True
 skip_missing_interpreters = True
+envlist = unit, sanity, integration
```
Comment (DnPlas) [inline persistenceagent/tox.ini:6]: Same as [this](https://github.com/canonical/pipelines-rocks/pull/63#discussion_r1452357669)

Comment (orfeas-k) [reply persistenceagent/tox.ini]: Updated

File: persistenceagent/tox.ini [hunk 2]
```diff
@@ -12,57 +13,61 @@ setenv =
...
+    ; git clone --branch {env:CHARM_BRANCH} {env:CHARM_REPO} {env:LOCAL_CHARM_DIR}
+    ; # upload rock to docker and microk8s cache, replace charm's container with local rock reference
+    ; bash -c 'NAME=$(yq eval .name rockcraft.yaml) && \
+    ;          VERSION=$(yq eval .version rockcraft.yaml) && \
+    ;          DOCKER_IMAGE=$NAME:$VERSION && \
+    ;          docker save $DOCKER_IMAGE > $DOCKER_IMAGE.tar && \
+    ;          sudo microk8s ctr image import $DOCKER_IMAGE.tar --digests=true && \
+    ;          yq e -i ".resources.oci-image.upstream-source=\"$DOCKER_IMAGE\"" {env:LOCAL_CHARM_DIR}/charms/kfp-persistence/metadata.yaml'
+    ; # run bundle integration tests with rock
+    ; tox -c {env:LOCAL_CHARM_DIR} -e bundle-integration-v2 -- --model kubeflow
```
Comment (misohu) [inline persistenceagent/tox.ini:73]: Same as [here](https://github.com/canonical/pipelines-rocks/pull/63#discussion_r1453085618).

**Review by misohu — APPROVED** [review]
